pkg/kine/drivers/generic/generic.go

```diff
@@ -381,6 +393,119 @@ func (d *Generic) queryRowPrepared(ctx context.Context, txName, sql string, prep
 	return r
 }
 
+func (d *Generic) InitializeWriteBatching(ctx context.Context) {
+	if d.BatchingEnabled && d.BatchingInterval > 0 && d.BatchingMaxQueries > 0 {
```

Comment:
We are adding a new feature, I would recommending validating the config and failing early if it is not proper (e.g. if batchingMaxQueries or batchingInterval are negative, let's shout the error instead of silently disabling this.

I would also like a logging message like this:

```go
logrus.WithFields(logrus.Fields{"interval": interval, "queue_size": maxQueries}).Print("Enable batching of insert queries")
```

or

```
logrus.Print("Disable batching of insert queries")
```

Reply:
Maybe also something similar to https://github.com/canonical/k8s-dqlite/blob/0ce314e0424d3e2cf744c64cd4e6d1b3d215cfb9/pkg/server/server.go#L253

Reply:
We also need an upper limit for the max-queue parameter. Otherwise we can run into 
```
time="2023-10-20T11:34:40+02:00" level=error msg="Process batch queue error" error="too many SQL variables"
```
when creating the query.
See also: https://www.sqlite.org/c3ref/c_limit_attached.html#sqlitelimitvariablenumber